juriscraper/scotus/scotus_docket_report.py

```diff
@@ -0,0 +1,413 @@
+
+        # Keep insertion order of first appearance of each PartyName
+        grouped: dict[str, list[dict[str, Any]]] = {}
+        order = []
+        for party in type_parties:
+            party_name = party.get("PartyName", "").strip()
+            if party_name not in grouped:
+                grouped[party_name] = []
+                order.append(party_name)
+            grouped[party_name].append(self._build_attorney(party))
+
+        return [
+            {
+                "type": type_key,
+                "name": party_name,
+                "attorneys": grouped[party_name],
+            }
+            for party_name in order
+        ]
+
+    @property
+    def parties(self) -> list[dict[str, Any]]:
+        """Return parties with their attorneys from Petitioner/Respondent arrays.
+
+            [
+               {
+                  "attorneys":[
+                     {
+                        "address":"...",
+                        "city":"Seattle",
+                        "email":"user@email.org",
+                        "is_counsel_of_record":true,
+                        "name":"...",
+                        "phone":"...",
+                        "state":"WA",
+                        "title":"...",
+                        "zip":"98101"
+                     }
+                  ],
+                  "name":"...",
+                  "type":"Petitioner"
+               }
+            ]
+
+        :return: List of parties dicts.
+        """
+        parties = []
+        parties.extend(self._build_parties_for_type("Petitioner"))
+        parties.extend(self._build_parties_for_type("Respondent"))
+        parties.extend(self._build_parties_for_type("Other"))
+        return parties
+
+    @classmethod
+    def normalize_date(
+        cls, date_str: Optional[str]
+    ) -> Optional[datetime.date]:
+        """Convert a date string to YYYY-MM-DD using common SCOTUS formats.
+
+        :param date_str: Raw date string (e.g., "July 3, 2024", "Jul 03 2024", "08/16/2024")
+        :return: A date python object or None
+        """
+        if not date_str:
+            return None
+        s = date_str.strip()
+        for date_format in cls.DATE_FORMATS:
+            try:
+                return datetime.strptime(s, date_format).date()
+            except ValueError:
+                continue
+        return None
+
+    @staticmethod
+    def _apply_prefix(prefix: str, parts: list[str]) -> list[str]:
+        """Prepend prefix to numeric parts."""
+        return [
+            f"{prefix}{p.strip()}" if p.strip().isdigit() else p.strip()
+            for p in parts
+        ]
+
+    @staticmethod
+    def _clean_parts(parts: list[str]) -> list[str]:
+        """Clean lower court numbers
+
+        :param parts: list of strings
+        :return: list of cleaned lower court numbers
+        """
+        return [p.strip() for p in parts]
+
+    def clean_lower_court_cases(self, raw_text: str) -> list[str]:
```

Comment:
we should refactor this function to use a single regex to handle all cases where there's a single prefix followed by a list of case numbers (case 2, 3, 5, 9 and 10). It's better to combine them so we don't have to repeat the same logic multiple times.

Reply:
Yeah, it was possible to merge those cases into a single one. I’ve refactored `clean_lower_court_cases` and added unit tests for the method.

thanks!